activemodel/lib/active_model/validations/acceptance.rb

```diff
@@ -17,54 +17,54 @@ def validate_each(record, attribute, value)
       private
 
         def setup!(klass)
-          klass.include(LazilyDefineAttributes.new(AttributeDefinition.new(attributes)))
+          klass.include LazilyDefineAttributes
```

Comment:
How about memoizing `@acceptance_attributes` here?

```suggestion
          klass.include LazilyDefineAttributes
          @acceptance_attributes =
            (LazilyDefineAttributes.acceptance_attributes_for(klass) + attributes).map!(&:to_s)
```

Reply:
Yeah, I know we could shove them somewhere in the class. But at that point, any application that happens to have defined `acceptance_attributes` (which is not unlikely) would potentially have some very strange bugs.

I don't really feel that there is such an issue here that we need to do something like this. At least for the "acceptance attributes", the first time *any* instance of a model hits method_missing, it will define the accessors and we should never hit `method_missing` again (for those attributes at least). So I think it's a relatively narrowly-scoped issue we're talking about.

Reply:
I have a totally different idea: what if instead of overriding `method_missing` and `respond_to_missing`, we override `initialize` on the model so that after the first model has been created, we define the accessors? This avoids preloading the schema and also does not incur any extra performance hit on method calls.

Reply:
Faster idea is welcome, since slower `respond_to?` is sometimes bothoring us.

For example cc2d614, the purpose of cc2d614 is to bypass `respond_to?(:empty?)` in `blank?`, since `respond_to?` on Active Record is much slower due to `DynamicMatchers`.

https://github.com/rails/rails/blob/533dd8a681a3ea7d12acdfc6a2d6a95891282b49/activesupport/lib/active_support/core_ext/object/blank.rb#L18-L20

https://github.com/rails/rails/blob/533dd8a681a3ea7d12acdfc6a2d6a95891282b49/activerecord/lib/active_record/dynamic_matchers.rb#L6-L13

Reply:
Anyway, I agree that it's a relatively narrowly-scoped issue in this PR.